ConvivaTestApp/src/androidTest/java/com/bitmovin/analytics/conviva/testapp/ConvivaAnalyticsTest.kt

```diff
@@ -0,0 +1,1570 @@
+        metadata.streamUrl = "CUSTOM_STREAM_URL"
+
+        return metadata
+    }
+
+    fun expectedContentMetadata(player: BitmovinPlayer,
+                                source: SourceItem,
+                                streamType: ContentMetadata.StreamType,
+                                duration: Int,
+                                overrideMetadata: MetadataOverrides,
+                                overrideCustom: Boolean = false) : ContentMetadata {
+        val contentMetadata = ContentMetadata()
+        contentMetadata.applicationName = overrideMetadata.applicationName
+        contentMetadata.viewerId = overrideMetadata.viewerId
+        contentMetadata.assetName = overrideMetadata.assetName
+        contentMetadata.duration = duration
+        contentMetadata.defaultResource = null
+        contentMetadata.encodedFrameRate = -1
+        contentMetadata.defaultBitrateKbps = -1
+        contentMetadata.isOfflinePlayback = false
+        contentMetadata.custom = HashMap()
+        overrideMetadata.custom.forEach { s, s2 ->
+            contentMetadata.custom[s] = s2
+        }
+
+        if (overrideCustom) {
+            contentMetadata.streamUrl = overrideMetadata.streamUrl ?: source.dashSource.url
+            contentMetadata.streamType = overrideMetadata.streamType ?: streamType
+            contentMetadata.custom["streamType"] = overrideMetadata.custom["streamType"] ?: source.type.name
+            contentMetadata.custom["integrationVersion"] = overrideMetadata.custom["integrationVersion"] ?: com.bitmovin.analytics.conviva.BuildConfig.VERSION_NAME
+        } else {
+            contentMetadata.streamUrl = source.dashSource.url
+            contentMetadata.streamType = streamType
+            contentMetadata.custom["streamType"] = source.type.name
+            contentMetadata.custom["integrationVersion"] = com.bitmovin.analytics.conviva.BuildConfig.VERSION_NAME
+        }
+        return contentMetadata
+    }
+
+    fun MockKMatcherScope.metadataEq(expectedMetatada: ContentMetadata) = match<ContentMetadata>  {
+        if (it.assetName == expectedMetatada.assetName &&
+            it.applicationName == expectedMetatada.applicationName &&
+            it.viewerId == expectedMetatada.viewerId &&
+            it.streamType == expectedMetatada.streamType &&
+            it.streamUrl == expectedMetatada.streamUrl &&
+            it.duration == expectedMetatada.duration &&
+            it.defaultResource == expectedMetatada.defaultResource &&
+            it.encodedFrameRate == expectedMetatada.encodedFrameRate &&
+            it.custom.equals(expectedMetatada.custom)) {
+            true
+        } else {
+            false
+        }
+    }
+
+    @Test
+    fun test_implicitSessionStartOnManualPlay() {
+        val activityScenario = ActivityScenario.launch(MainActivity::class.java)
+        activityScenario.onActivity { activity: MainActivity ->
+            try {
```

Comment:
Do we really need this in every test? 🤔 looks like tedious boilerplate code. Any chance to get rid of this? Would it be possible to move the `activity` handling into `createConvivaAnalyticsObject`?

Reply:
Removed most of boiler plate code